src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -808,6 +806,12 @@ EnsureTableCanBeColocatedWith(Oid relationId, char replicationModel,
 								  "for hash distributed tables.")));
 	}
 
+	char sourceReplicationModel = sourceTableRef->cacheEntry->replicationModel;
+	Oid sourceDistributionColumnType =
+		sourceTableRef->cacheEntry->partitionColumn->vartype;
+
+	ReleaseTableCacheEntry(sourceTableRef);
```

Comment:
What is the implication of not calling `ReleaseTableCacheEntry ` ?

Reply:
Reference will be freed when memory context is destroyed